### PR TITLE
[hmac, sw] Revert hash stop hang workaround

### PR DESCRIPTION
**Pull request by andrea-caforio**
In line with the RTL fix in #24771 for the hash stop hang bug described in #24767, this commit reverts four commits that implemented a software workaround for the issue and only keeps the idle polling timeout implementation from #24944. This workaround and its corresponding documentation is now obsolete.

Reverted commits:
  - e869152 #25070
  - 2b072c4 #25012
  - 7f32449 #24966
  - f591bea #24944

### CODE REVIEW

File: sw/device/lib/crypto/drivers/hmac.c [hunk 1]
```diff
@@ -89,6 +75,8 @@ enum {
  * It returns error if HMAC HWIP becomes idle without firing `hmac_done`
  * interrupt.
  *
+ * TODO(#23191): It might be beneficial to have a timeout value for the polling.
+ *
```
Comment (vogelpi) [inline sw/device/lib/crypto/drivers/hmac.c:79]: Doing a plain revert of the other commits is fine and the approach is good, it's easier to review this.

What is not ideal is that it means to re-introduce this TODO here which got resolved by the workaround. Would it be possible to do a separate commit or even PR (whatever you prefer @andrea-caforio ) to resolve this? You can take inspiration from the reverted commits on how to do this.

Comment (andrea-caforio) [reply sw/device/lib/crypto/drivers/hmac.c]: Thank you @vogelpi for the review and catching this issue. I reinstantiated the polling timeout implementation in a separate commit as to not spam the PR list with code that has already been review before.

**Review by vogelpi — APPROVED** [review]
This looks good to me, thanks @andrea-caforio !

**Review by martin-velay — APPROVED** [review]
This LGTM, thanks @andrea-caforio for taking care of it !

**Review by jadephilipoom — APPROVED** [review]
